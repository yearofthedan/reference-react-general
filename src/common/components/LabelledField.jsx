import React from "react";
import PropTypes from "prop-types";

const LabelledField = ({ id, type, value, onChange, children }) => (
    <>
        <label htmlFor={id}>
            {children}
        </label>
        <input type={type} value={value} onChange={onChange} id={id} />
    </>
);

LabelledField.propTypes = {
    id: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    type: PropTypes.string
};

LabelledField.defaultValues = {
    type: 'text'
};

export default LabelledField;

