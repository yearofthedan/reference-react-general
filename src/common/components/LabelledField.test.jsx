import React from "react";
import LabelledField from "./LabelledField";
import renderer from "react-test-renderer";

describe('<LabelledField />', () => {
   it('renders to its snapshot', () => {
       const tree = renderer
           .create(
               <LabelledField id="some-id" onChange={jest.fn()}  value="some-value" type="number">
                   Some label
               </LabelledField>
           )
           .toJSON();
       expect(tree).toMatchSnapshot();
   });
});
