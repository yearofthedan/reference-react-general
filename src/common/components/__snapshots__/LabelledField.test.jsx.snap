// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelledField /> renders to its snapshot 1`] = `
Array [
  <label
    htmlFor="some-id"
  >
    Some label
  </label>,
  <input
    id="some-id"
    onChange={[MockFunction]}
    type="number"
    value="some-value"
  />,
]
`;
